libs/@local/hql/cst/src/parse.rs

```diff
@@ -0,0 +1,119 @@
+use core::mem;
+
+use winnow::{
+    ascii::multispace0,
+    combinator::{delimited, separated_foldl1, trace},
+    error::ParserError,
+    stream::{AsChar, Stream, StreamIsPartial},
+    Parser,
+};
+
+use crate::arena::{self, Arena};
+
+enum VecOrOneValue<'a, T> {
+    Vec(arena::Vec<'a, T>),
+    One(T),
+    None,
+}
```

Comment:
I don't really mind as this is internal, but I'd prefer `Option<Vec | One>` instead of `Vec | One | None`

Reply:
rewritten in [241f0ee](https://github.com/hashintel/hash/pull/4753/commits/241f0ee12b86d2a3331bcefac4883f3b662f6bcc)

Reply:
Much better 🙂